web/server/api/endpoints/commands.py

```diff
@@ -92,8 +103,20 @@ async def evaluate(
             limit=options.limit,
         )
     except Exception:
+        error_type, error_value, error_traceback = sys.exc_info()
+
         raise HTTPException(
-            status_code=HTTP_422_UNPROCESSABLE_ENTITY, detail=traceback.format_exc()
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=Error(
+                timestamp=now_timestamp(),
+                status=HTTP_422_UNPROCESSABLE_ENTITY,
+                message="Unable to evaluate model",
+                origin="API -> commands -> evaluate",
+                description=str(error_value),
+                type=str(error_type),
+                traceback=traceback.format_exc(),
+                stack=traceback.format_tb(error_traceback),
+            ).dict(),
```

Comment:
I think you might be able to DRY this up by using FastAPI's default exception handler. If you want to add the custom message and origin fields, maybe you can create a custom exception that take those as parameters and re-raise it.
https://fastapi.tiangolo.com/tutorial/handling-errors/#override-the-default-exception-handlers